### PR TITLE
Feature/persistant annotations

### LINKED ISSUE #57: Re-use columns selected last time
Use the persisted config storage to re-load columns from last time, this may already be done... is it working...?

### PR DESCRIPTION
**Pull request by BrianWhitneyAI**
### Description of Changes 

This PR proposes prioritizing recently used Groups, Filters, and Sorts in the ordering of their respective selection dropdowns ( resolves #57 ). This PR is a continuation of #76  feature/recent-hierarchy-annotations , but is based on the new generalize-ui #79 branch.

### Testing 

Monitored state through various selections and deselections of Groups, Filters, and Sorts including those with repeated no-file-found options.

I did experience some issues with reloading the application where it would disconnect on the first reload but succeed on the second reload. I believe this is an inherent issue to the base branch or my development environment as is still present without these changes.

This PR also passes tests locally with the exception of 
```
  9) @renderer FileDownloadServiceElectron
       promptForDownloadDirectory
         complains about non-writeable directory when given:
     Error: Trying to stub property 'invoke' of undefined

```
I believe that this is unique to my local environment since I had to use WSL to run the full test suite. In reviewing this PR it would be super helpful to run tests locally with your dev env just to make sure everything is Ok.

### Possible Changes/Adjustments 

1) Limiting the length of recent annotations to a smaller number, I foresee a situation where a user has a lot of interaction and the recent list becomes very long (and unalphabetized) and this feature becomes a hindrance 
2) A adjustment to ‘On Start” state to not include undefined elements.

### CODE REVIEW

**Comment by BrianWhitneyAI** [comment]
https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/81#discussion_r1592953711

**Comment by BrianWhitneyAI** [comment]
https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/81#discussion_r1592956052

**Comment by BrianWhitneyAI** [comment]
https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/81#discussion_r1592957387

**Comment by BrianWhitneyAI** [comment]
https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/81#discussion_r1592960258

**Comment by BrianWhitneyAI** [comment]
https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/81#discussion_r1592963169

**Comment by BrianWhitneyAI** [comment]
https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/81#discussion_r1592963734

**Comment by BrianWhitneyAI** [comment]
https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/81#discussion_r1592965241

**Comment by BrianWhitneyAI** [comment]
https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/81#discussion_r1592969865

**Comment by BrianWhitneyAI** [comment]
https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/81#discussion_r1592970276

File: packages/core/components/AnnotationPicker/index.tsx [hunk 1]
```diff
@@ -4,8 +4,9 @@ import { useSelector } from "react-redux";
...
 import { ListItem } from "../ListPicker/ListRow";
 import { TOP_LEVEL_FILE_ANNOTATION_NAMES } from "../../constants";
-import Annotation from "../../entity/Annotation";
-import { metadata, selection } from "../../state";
+import Annotation, { AnnotationName } from "../../entity/Annotation";
+import { selection } from "../../state";
+import { uniqBy } from "lodash";
 
 interface Props {
     disabledTopLevelAnnotations?: boolean;
```
Comment (aswallace) [inline packages/core/components/AnnotationPicker/index.tsx:12]: Line 14: outside the scope of this PR, but I'm unsure if the `disableUnavailableAnnotations` prop still does anything?

**Review by aswallace — APPROVED** [review]
Ran tests locally and they worked fine for me! Left a few comments

> 1. Limiting the length of recent annotations to a smaller number, I foresee a situation where a user has a lot of interaction and the recent list becomes very long (and unalphabetized) and this feature becomes a hindrance

I'm not quite sure what you mean by this one. I did quite a bit of interaction locally and the recent list stayed limited, especially since the non-selected recents are hidden when other filters are selected (e.g., if 3 filters are selected then only 2 recents show because of the slicing)

File: packages/core/components/ListPicker/index.tsx [hunk 2]
```diff
@@ -160,7 +160,8 @@ export default function ListPicker(props: ListPickerProps) {
             </div>
             <div className={styles.footer}>
                 <h6>
-                    Displaying {filteredItems.length} of {items.length} Options
+                    {/* (item.lenght -1) to account for buffer in item list. */}
```
Comment (kmitcham) [inline packages/core/components/ListPicker/index.tsx:163]: nit: length

File: packages/core/components/ListPicker/ListRow.tsx [hunk 3]
```diff
@@ -59,6 +62,7 @@ export default function ListRow(props: Props) {
                 <div>{item.selected && <Icon iconName="CheckMark" />}</div>
                 {item.displayValue}
             </label>
+            {item.recent && <Icon iconName="Redo" />}
```
[the remark this answers is not in the document]
Comment (BrianWhitneyAI) [reply packages/core/components/ListPicker/ListRow.tsx:65]: @SeanLeRoy @aswallace Im not too familiar with the Icon library. This was mostly a placeholder. Is there a different icon we want to use here?

**Review by SeanLeRoy — APPROVED** [review]
I think resolving Anya's comments and getting File Name moved would make this good to go to me

**Comment by BrianWhitneyAI** [comment]
closing in favor of #110

**Pull request closed without merging**